kornia/nerf/rays.py

```diff
@@ -0,0 +1,427 @@
+import math
+from typing import Dict, List, Optional, Tuple
+
+import torch
+
+from kornia.core import Device, Tensor
+from kornia.geometry.camera import PinholeCamera
+from kornia.nerf.camera_utils import cameras_for_ids
+from kornia.utils._compat import torch_meshgrid
+from kornia.utils.helpers import _torch_inverse_cast
+
+
+class RaySampler:
```

Comment:
as pointed above, some parts of this class could be abstracted as a generic Ray class